nixtla/nixtla_client.py

```diff
@@ -995,7 +1036,15 @@ def forecast(
             )
             in_sample_df = ufp.drop_columns(in_sample_df, target_col)
             out = ufp.vertical_concat([in_sample_df, out])
+        self._maybe_assign_feature_contributions(
+            feature_contributions=resp["feature_contributions"],
+            x_cols=x_cols,
+            out_df=out[[id_col, time_col, "TimeGPT"]],
+            insample_feat_contributions=insample_feat_contributions,
+        )
+        if add_history:
             out = ufp.sort(out, by=[id_col, time_col])
+
```

Comment:
For consistency I think we should also sort the feature contributions, so maybe something like this:
```suggestion
        if add_history:
            sort_idxs = ufp.maybe_compute_sort_indices(out, id_col=id_col, time_col=time_col)
            assert sort_idxs is not None
            out = ufp.take_rows(out, sort_idxs)
            self.feature_contributions = ufp.take_rows(self.feature_contributions, sort_idxs)
```